### PR TITLE
Implement find/replace in .Net Core editing

### PR DESCRIPTION
**Pull request by zur003**
Working on #6139

### CODE REVIEW

File: ApsimNG/Utility/FindAndReplaceForm.cs [hunk 1]
```diff
@@ -69,17 +72,37 @@ private void Window1_Destroyed(object sender, EventArgs e)
 
         public void Destroy()
         {
+#if NETCOREAPP
+            window1.Dispose();
+#else
             window1.Destroy();
+#endif
```
Comment (hol430) [inline ApsimNG/Utility/FindAndReplaceForm.cs:79]: This conditional code could be replaced with a call to `window1.Cleanup()`, which is an extension method in the `UserInterface.Extensions` namespace:

```suggestion
            window1.Cleanup();
```

File: ApsimNG/Utility/FindAndReplaceForm.cs [hunk 2]
```diff
@@ -69,17 +72,37 @@ private void Window1_Destroyed(object sender, EventArgs e)
...
+#if NETCOREAPP
+        SearchContext context;
+        SearchContext Context
+        {
+            get { return context; }
+            set { context = value; }
+        }
+
+        SourceView editor;
+        SourceView Editor
```
Comment (hol430) [inline ApsimNG/Utility/FindAndReplaceForm.cs:100]: Could we have some explicit accessibility modifiers here? We also probably don't need to have a manual backing field for the Context property.

```suggestion
#if NETCOREAPP
        public SearchContext Context { get; set; }

        private SourceView editor;
        public SourceView Editor
```

File: ApsimNG/Utility/FindAndReplaceForm.cs [hunk 3]
```diff
@@ -88,26 +111,67 @@ TextEditor Editor
...
         /// </summary>
         public void ShowMsg(string message)
         {
             MessageDialog md = new MessageDialog(Editor.Toplevel as Window, DialogFlags.Modal, MessageType.Info, ButtonsType.Ok, message);
             md.Run();
+#if NETCOREAPP
+            md.Dispose();
+#else
             md.Destroy();
+#endif
```
Comment (hol430) [inline ApsimNG/Utility/FindAndReplaceForm.cs:126]: This conditional code could be replaced with a call to `md.Cleanup()` (in the `UserInterface.Extensions` namespace).

**Comment by hol430** [comment]
Thanks Eric - looks good. This will be a good addition to the .net core gui. I think it's one of the few features which haven't been brought across yet.

File: ApsimNG/Utility/FindAndReplaceForm.cs (hunk 1 above)
Comment (zur003) [reply ApsimNG/Utility/FindAndReplaceForm.cs]: Ah, right. I thought there was something like that.

**Comment by zur003** [comment]
I also fixed a bug in which repeated undos would totally clear the editing buffer.

One problem remains - use of ctrl-f and  ctrl-h accelerator keys isn't set up until the first time the popup is populated, so if you start editing and press crtl-f, nothing happens. After you've activated the popup menu once, it works as expected.

**Pull request merged**